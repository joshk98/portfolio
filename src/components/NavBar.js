import React from "react";
import { NavLink } from "react-router-dom";

import "../styles/navbar.css";

function NavBar() {
  return (
    <nav className="navbar-container">
      <ul className="navbar-container-list">
        <li className="navbar-container-list-item">
          <NavLink to="/">About Me</NavLink>
        </li>
        <li className="navbar-container-list-item">
          <NavLink to="/projects">Projects</NavLink>
        </li>
        <li className="navbar-container-list-item">
          <NavLink to="/contactme">Contact Me</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default NavBar;
