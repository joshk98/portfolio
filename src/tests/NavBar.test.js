import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
