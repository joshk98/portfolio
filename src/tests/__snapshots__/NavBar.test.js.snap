// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar-container"
  >
    <ul
      class="navbar-container-list"
    >
      <li
        class="navbar-container-list-item"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          About Me
        </a>
      </li>
      <li
        class="navbar-container-list-item"
      >
        <a
          class=""
          href="/projects"
        >
          Projects
        </a>
      </li>
      <li
        class="navbar-container-list-item"
      >
        <a
          class=""
          href="/contactme"
        >
          Contact Me
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
